src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -2433,61 +2435,16 @@ public String restrictSelectedFiles(boolean restricted) throws CommandException{
         return  returnToDraftVersion();
     }
 
-    public void restrictFiles(boolean restricted) throws CommandException {
-   
-        //if (previouslyRestrictedFiles == null) {
-        // we don't need to buther with this "previously restricted" business 
-        // when in Create mode... because all the files are new, so none could 
-        // have been restricted previously;
-        // (well, it looks like the code below should never be called in the 
-        // CREATE mode in the first place... the edit files fragment uses
-        // its own restrictFiles() method there; also, the fmd.getDataFile().equals(fmw.getDataFile()))
-        // line is not going to work on a new file... so be mindful of all this
-        // when the code between the 2 beans is merged in 4.3.
-        if (editMode != EditMode.CREATE) {
-            previouslyRestrictedFiles = new ArrayList<>();
-            for (FileMetadata fmd : workingVersion.getFileMetadatas()) {
-                if (fmd.isRestricted()) {
-                    previouslyRestrictedFiles.add(fmd);
-                }
+    private void restrictFiles(boolean restricted) throws CommandException {
+        Command<Void> cmd;
+        previouslyRestrictedFiles = new ArrayList<>();
+        for (FileMetadata fmd : this.getSelectedFiles()) {
+            if(fmd.isRestricted()) {
+                previouslyRestrictedFiles.add(fmd);
             }
-            
-            Command cmd;
-            String fileNames = null;
-            for (FileMetadata fmw : workingVersion.getFileMetadatas()) {
-                for (FileMetadata fmd : this.getSelectedFiles()) {
-                    if (restricted && !fmw.isRestricted()) {
```

Comment:
What this loop does that the replacement above doesn't is to handle the case where a file is, for example, restricted in the released version and already unrestricted in the draft and the user wants to unrestrict it. I don't think there's any real effect of letting this happen (an unrestricted file being set to unrestricted), but if this case needs to still be caught, it can be caught more efficiently in the command itself (which has to find the FileMetadata for this file that is associated with the draft version anyway), and/or could be caught when deciding if the selected files are all valid (it would be inefficient to do it there, but if the user needs to be alerted (versus the current mode of silently doing a noop), it might be better to completely remove the selection and edit items from a display of released versions when a draft version exists ("Edits can be made to the existing draft(link) version"). (Similar issue if the restrict request is from a published version and the file has already been removed from an existing draft. In the existing code and this PR (after second commit), there is silently no change made for this case.)